bin/workflows/pycbc_make_faithsim_workflow

```diff
@@ -0,0 +1,187 @@
+            out_dir=out_dir,
+            tags=[tag] + tags,
+        ).create_node()
+        node.add_input_opt("--input-file", input_file)
+        node.new_output_file_opt(analysis_time, ".png", "--output-plot")
+        workflow += node
+        files += node.output_files
+
+    return files
+
+
+class CreateInjectionExecutable(wf.Executable):
+    """Class for running create_inj_file"""
+
+    current_retention_level = wf.Executable.ALL_TRIGGERS
+
+    def create_node(self, analysis_time, extra_tags=None):
+        if extra_tags is None:
+            extra_tags = []
+
+        node = wf.Executable.create_node(self)
+
+        node.new_output_file_opt(
+            analysis_time, ".xml", "--output", tags=self.tags + extra_tags
+        )
+        return node
+
+
+class FaithsimExecutable(wf.Executable):
+    """Class for running pycbc_faithsim"""
+
+    current_retention_level = wf.Executable.ALL_TRIGGERS
+
+    def create_node(self, analysis_time, param_file, extra_tags=None):
+        if extra_tags is None:
+            extra_tags = []
+        node = wf.Executable.create_node(self)
+
+        node.add_input_opt("--param-file", param_file)
+
+        node.new_output_file_opt(
+            analysis_time, ".dat", "--match-file", tags=self.tags + extra_tags
+        )
+        return node
+
+
+class CollectResultsExecutable(wf.Executable):
+    """Class for collecting the results of the faithsim script"""
+
+    current_retention_level = wf.Executable.ALL_TRIGGERS
+
+    def create_node(self, analysis_time, match_list, bank_list, extra_tags=None):
+        if extra_tags is None:
+            extra_tags = []
+        node = wf.Executable.create_node(self)
+
+        for match in match_list:
+            node.add_input_opt("--inputs-match", match)
+        for bank in bank_list:
+            node.add_input_opt("--inputs-bank", bank)
+
+        node.new_output_file_opt(
+            analysis_time, ".dat", "--output", tags=self.tags + extra_tags
+        )
+
+        return node
+
+
+parser = argparse.ArgumentParser(description=__doc__)
+parser.add_argument("--version", action="version", version=__version__)
+wf.add_workflow_command_line_group(parser)
+wf.add_workflow_settings_cli(parser)
+args = parser.parse_args()
+
+workflow = wf.Workflow(args)
+
+config_path = os.path.abspath(args.config_files[0])
+
+wf.makedir(args.output_dir)
+os.chdir(args.output_dir)
+
+wf.makedir("match")
+wf.makedir("collect_results")
+wf.makedir("injections")
+wf.makedir("bank")
+wf.makedir("plots")
+
+num_banks = workflow.cp.get("splitbank", "num_banks")
+
+injections_exe = CreateInjectionExecutable(
+    workflow.cp,
+    "lalapps_inspinj",
+    ifos=["H1", "L1", "V1"],
+    out_dir="injections",
+    tags=["inj"],
+)
+
+inj_node = injections_exe.create_node(workflow.analysis_time)
+workflow += inj_node
+
+inj = inj_node.output_files[0]
+
+split_exe = wf.PycbcSplitBankXmlExecutable(
+    workflow.cp, "pycbc_splitbank", num_banks=num_banks, out_dir="bank"
```

Comment:
If removing lines 116 - 123, this may then need to use workflow.output_dir + "bank" (or similar if it's not output_dir but something else) to make this an absolute path.